### PR TITLE
Backport Kinesis Enhancements

### PR DESCRIPTION
**Pull request by danotorrey**
Backport AWS Kinesis/CloudWatch feature (#247) to 3.1

Please complete the following steps to review this PR:

1) Check out this branch in a 3.1 checkout. Make sure the server starts successfully without error. 
2) Make sure you can add the AWS Kinesis/CloudWatch input from the System > Inputs page.
3) Make sure it looks like all code has been included from original PR.
4) On the UI side, spot check to make sure the merges look good. Backporting caused quite a lot of merge conflicts since this was originally coded for release 2.2. Compare the changes to the original PR #247 for reference.

### CODE REVIEW

**Comment by danotorrey** [comment]
@kyleknighted May I please have your help backporting the AWS Kinesis Enhancements to the `3.1` branch? This PR is a start to that. 

I need some help merging the UI changes. I am wondering if you can check out this branch and cherry-pick the enhancements and resolve the UI merge conflicts `git cherry-pick c3301d843ff8828b457c41505064ad6f7d51a616`. The `pom.xml` is the only other conflicting file, and I have fixed it already in this branch, so you can ignore changes in that file when doing the cherry pick.

**Comment by danotorrey** [comment]
@ceruleancee @lingpri This backport is ready for review. Kyle and I were able to resolve the UI issues.

**Comment by danotorrey** [comment]
@ceruleancee @lingpri Please ignore the linter failure. This is a known issue and should not prevent the merge of this PR.

File: src/main/java/org/graylog/integrations/aws/resources/AWSResource.java [hunk 1]
```diff
@@ -100,35 +100,26 @@ public KinesisPermissionsResponse getPermissions() {
...
     @POST
     @Timed
     @Path("/cloudwatch/log_groups")
     @ApiOperation(value = "Get all available AWS CloudWatch log groups names for the specified region.")
     @RequiresPermissions(AWSPermissions.AWS_READ)
     @NoAuditEvent("This does not change any data")
-    public LogGroupsResponse getLogGroupNames(@ApiParam(name = "JSON body", required = true) @Valid @NotNull AWSRequestImpl awsRequest) {
-        return cloudWatchService.getLogGroupNames(awsRequest.region(), awsRequest.awsAccessKeyId(), awsRequest.awsSecretAccessKey());
+    public LogGroupsResponse getLogGroupNames(@ApiParam(name = "JSON body", required = true) @Valid @NotNull AWSRequestImpl request) {
+        return cloudWatchService.getLogGroupNames(request);
```
Comment (lingpri) [inline src/main/java/org/graylog/integrations/aws/resources/AWSResource.java]: @danotorrey is this a new method , that is able to extract all the information from the requeust ?

File: pom.xml [hunk 2]
```diff
@@ -32,8 +32,9 @@
         <maven.deploy.skip>true</maven.deploy.skip>
         <maven.site.skip>true</maven.site.skip>
         <graylog.version>3.1.4-SNAPSHOT</graylog.version>
-        <aws-java-sdk-2.version>2.7.33</aws-java-sdk-2.version>
-        <aws-kinesis-client.version>2.2.2</aws-kinesis-client.version>
+        <aws-java-sdk-2.version>2.10.26</aws-java-sdk-2.version>
+        <aws-kinesis-client.version>2.2.6</aws-kinesis-client.version>
+        <integrations.protobuf.version>3.11.1</integrations.protobuf.version>
```
Comment (lingpri) [inline pom.xml]: @danotorrey is this 3.11.1 or 3.1 ?

File: src/main/java/org/graylog/integrations/aws/resources/requests/AWSInputCreateRequest.java [hunk 3]
```diff
@@ -76,20 +60,38 @@
...
-                                               @JsonProperty(ADD_FLOW_LOG_PREFIX) boolean addFlowLogPrefix) {
-        return new AutoValue_AWSInputCreateRequest(name, awsMessageType, awsAccessKey, awsSecretKey,
-                                                   streamName, assumeRoleArn, region, batchSize, global,
-                                                   enableThrottling, addFlowLogPrefix);
+    public static Builder builder() {
+        return Builder.create();
+    }
+
+    @AutoValue.Builder
+    public static abstract class Builder implements AWSRequest.Builder<Builder> {
```
Comment (lingpri) [inline src/main/java/org/graylog/integrations/aws/resources/requests/AWSInputCreateRequest.java]: @danotorrey Where would a Builder like this come in handy ? Can I re-purpose this builder else where within ipfixcodec.java to create objects of InformationElementDefinitions .java ?

**Comment by thll** [comment]
@danotorrey and I tested the setup flow end to end and also verified that messages are received. Looked all good!

**Pull request merged**